core/src/main/java/org/jsmart/zerocode/core/report/ZeroCodeReportGeneratorImpl.java

```diff
@@ -82,6 +81,26 @@ public ZeroCodeReportGeneratorImpl(ObjectMapper mapper) {
         this.mapper = mapper;
     }
 
+    /**
+     * gets unique steps from a scenario. In case of retries, the steps have same correlation id and if
+     * one of the retries is successful, we include it in the result(not the one which failed).
+     * @param steps
+     * @return
+     */
+    private List<ZeroCodeReportStep> getUniqueSteps(List<ZeroCodeReportStep> steps){
```

Comment:
Thanks! Looks great!

Can you add some unit tests for this `getUniqueSteps(...)` please?
That's covering some RETRY and non-RETRY cases?